source/extensions/filters/listener/proxy_protocol/proxy_protocol_header.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace ProxyProtocol {
+
+//
+// See https://github.com/haproxy/haproxy/blob/master/doc/proxy-protocol.txt
+// for definitions
+
+constexpr char PP1_SIGNATURE[] = "PROXY ";
+constexpr uint32_t PP1_SIGNATURE_LEN = 6;
+constexpr char PP2_SIGNATURE[] = "\x0d\x0a\x0d\x0a\x00\x0d\x0a\x51\x55\x49\x54\x0a";
+constexpr uint32_t PP2_SIGNATURE_LEN = 12;
+constexpr uint32_t PP2_HEADER_LEN = 16;
+constexpr uint32_t PP2_VERSION = 0x2;
+constexpr uint32_t PP2_ONBEHALF_OF = 0x1;
+constexpr uint32_t PP2_LOCAL = 0x0;
+
+constexpr uint32_t PP2_AF_INET = 0x1;
+constexpr uint32_t PP2_AF_INET6 = 0x2;
+constexpr uint32_t PP2_AF_UNIX = 0x3;
+
+struct PpHeader {
```

Comment:
nit: Given this is in the ProxyProtocol namespace, is "Pp" needed? Perhaps just "Header" or "WireHeader" ?